docker/daemon.go

```diff
@@ -264,6 +249,22 @@ func (cli *DaemonCli) CmdDaemon(args ...string) error {
 		"graphdriver": d.GraphDriver().String(),
 	}).Info("Docker daemon")
 
+	api := apiserver.New(serverConfig)
```

Comment:
there is actually a reason. The `ServeAPI` method initializes the http mux that handles all the connections.

The way it worked before this change was all static, the router was created before the daemon, the daemon was assigned to the server and everything worked before there was only a happy path.

The way it works after this change, the router creation is delegated until the last possible moment, inside that `ServerAPI`, so we can modify the routes the server is going to accept, in `InitRouters`.

As far as I can tell after my tests, everything keeps working as expected, because we really don't serve connections until we call `AcceptConnections`, so moving this code here is not sensitive.